rfd/0080-piv-private-key.md

```diff
@@ -0,0 +1,390 @@
+serial_number=<serial_number>
+slot=<slot>
+-----END YUBIKEY PIV PRIVATE KEY-----
+```
+
+#### Supported clients
+
+`tsh` and Teleport Connect will both support hardware private key login, and `tctl` will be able to use resulting login sessions. 
+
+#### Unsupported clients
+
+The WebUI will not be able to support PIV login, since it is brower-based and cannot connect directly to the user's PIV device. If a user with `require_session_mfa: hardware_key` attempts to login on the WebUI, or use an existing login session, it will fail. However, WebUI user registration and password reset logic must still work, regardless of the user's private key policy requirement. After intial registration/reset flow, the user should be directed to a page which notifies them that `tsh` or Teleport Connect must be used.
+
+It may be possible to work around this limitation by introducing a local proxy to connect to the hardware key, or by supporting a hardware key solution which doesn't need a direct connection, but this is out of scope and will not be explored in this PR. 
+
+In cases where WebUI access is needed or desired, cluster admins should only apply `require_session_mfa: hardware_key | hardware_key_touch` selectively to roles which warrant more protection. Teleport Connect will also serve as a great UI alternative.
+
+### UX
+
+#### Hardware key login
+
+When possible, hardware key login will not be any different from the normal login flow. However, in some cases, additional user intervention will be required. Below are some examples along with the resulting UX.
+
+Note: Teleport Connect will need custom solutions for these edge cases, such as tsh-initiated callbacks.
+ 
+##### Initial login fails due to an unmet private key policy
+
+```
+> tsh login --user=dev
+Enter password for Teleport user dev:
+Tap any security key
+Initial login failed due to an unmet private key policy, "hardware_key".
+Re-initiaiting login with YubiKey generated private key...
+Enter password for Teleport user dev:
+Tap any security key
+```
+
+Note: this should only occur when a user's role determines it's private key policy requirement, and the user does not have an existing login session which meets the required policy (expired or active).
+
+##### User's YubiKey not connected during login
+
+```
+> tsh login --user=dev
+Cluster "root" requires a YubiKey generated private key to login, but there
+is no YubiKey connected. Please insert a YubiKey to re-initiate login...
+// tsh polls the PIV library until the user connects a YubiKey (30 second timeout) or the user cancels
+Re-initiaiting login with YubiKey generated private key.
+Enter password for Teleport user dev:
+Tap any security key
+```
+
+##### User's Yubikey not connected during a request
+
+```
+> tsh ls
+Please insert the YubiKey used during login (serial number XXXXXX) to continue...
+// tsh polls the PIV library until the user connects a YubiKey (30 second timeout) or the user cancels
+```
+
+##### Touch requirement
```

Comment:
From a security and usability standpoint this should probably timeout after a few seconds

Reply:
This is by no means necessary, but I think it would improve UX a bit. Instead of providing an error to the user and prompting them to relogin, we can provide them with the chance to connect their hardware key and continue from where they are.

For example, in the case where the user tries to login with a soft key (default), but then `tsh` finds that they need to use a hardware key, this usually requires the user to provide their login credentials twice. If the user doesn't have their hardware key connected at the time, then the user would be given an error to connect their hardware key and they'd need to re-login. This would go through the same flow of using a soft key first before discovering a hardware key is required, resulting in the user being prompted for login credentials 3 times. This small UX improvement would reduce that to 2 times and speed up the process overall.

As for the security concern, I'm not sure exactly what you're referring to. The PIV card must be connected _before_ the user provides their login credentials, so there is no risk of a privileged session being taken during this 30 second window. let me know if you are referring to a different security concern.